core/authority_discovery/query/audi_store_impl.cpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "authority_discovery/query/audi_store_impl.hpp"
+
+#include <scale/kagome_scale.hpp>
+
+namespace kagome::authority_discovery {
+
+  AudiStoreImpl::AudiStoreImpl(std::shared_ptr<storage::SpacedStorage> storage)
+      : space_{storage->getSpace(storage::Space::kAudiPeers)} {
+    BOOST_ASSERT(space_ != nullptr);
+  }
+
+  void AudiStoreImpl::store(const primitives::AuthorityDiscoveryId &authority,
+                            const AuthorityPeerInfo &data) {
+    auto encoded = scale::encode(data);
+
+    if (not encoded) {
+      SL_ERROR(log_, "Failed to encode PeerInfo");
+      return;
+    }
+
+    if (auto res = space_->put(authority, common::Buffer(encoded.value()));
+        not res) {
+      SL_ERROR(
+          log_, "Failed to put authority {} error {}", authority, res.error());
+    }
+  }
+
+  std::optional<AuthorityPeerInfo> AudiStoreImpl::get(
+      const primitives::AuthorityDiscoveryId &authority) const {
+    auto res = space_->tryGet(authority);
+
+    // check if there was any critical error during reading
+    if (not res) {
+      SL_CRITICAL(log_,
+                  "Failed to get authority {} due to database error {}",
+                  authority,
+                  res.error());
+      return std::nullopt;
+    }
+
+    // check if the authority was not found
+    if (not res.value()) {
+      return std::nullopt;
+    }
+
+    auto decoded = scale::decode<AuthorityPeerInfo>(res.value().value());
+
+    if (not decoded) {
+      SL_ERROR(log_, "Failed to decode PeerInfo");
+      return std::nullopt;
+    }
+
+    return decoded.value();
+  }
+
+  bool AudiStoreImpl::remove(
```

Comment:
Return value is not used.
`remove` always returns `outcome::success`, even for non-existing key
```suggestion
  void AudiStoreImpl::remove(
```